opt_adapt/opt.py

```diff
@@ -632,4 +634,4 @@ def minimise(
 
         # Clean up
         tape.clear_tape()
-    return u_plus, mesh
+    return u_plus, fd.Mesh(mesh.coordinates.copy(deepcopy=True))
```

Comment:
If the copied mesh is stored in the `OptimisationProgress` class then you no longer need to output it here.